starter-code/src/viking.js

```diff
@@ -1,11 +1,97 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+Soldier.prototype.attack = function(){
+  return this.strength;
+}
+Soldier.prototype.receiveDamage = function(damage){
+  this.health -= damage;
+}
 
 // Viking
-function Viking() {}
+
+function Viking(name, health, strength) {
+  Soldier.call(this, health, strength);
+  this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+
+
+Viking.prototype.receiveDamage = function(damage){
+  this.health -= damage;
+  if (this.health !== 0) {
+    return this.name + " has received " + damage + " points of damage";
+  } else {
+    return this.name + " has died in act of combat";
+  }
+}
+
+
+Viking.prototype.battleCry = function(){
+  return "Odin Owns You All!";
+  
+}
+
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+  Soldier.call(this, health, strength);
+}
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
+
+Saxon.prototype.receiveDamage = function(damage){
+  this.health -= damage;
+  if (this.health !== 0) {
+    return "A Saxon has received " + damage + " points of damage";
+  } else {
+    return "A Saxon has died in combat";
+  }
+}
```

Comment:
Esta función os da error en función de si cargais todos los tests o cargais solo el concreto de la función que comprueba si el sajón muere. Esto pasa por la comparación que hacéis. `this.health !== 0` no es demasiado correcto, porque si le acabo quitando más vida de la que tiene, me dirá que está vivo, ya que `-1` es diferente a `0`. Es mejor si `(this.health <= 0) return "A Saxon has died in combat"`. Con los vikingos sí os salía bien seguramente porque Jasmine no acumulaba tanto daño hacia el vikingo como para quitarle más vida de la que tiene en su totalidad